pkg/os/shell/shell.go

```diff
@@ -208,7 +210,9 @@ func detectShellByCheckingProcessTree(p AbstractProcess) string {
 		if err != nil {
 			return ""
 		}
-		if processName == "zsh" || processName == "bash" || processName == "fish" {
+		if crcstrings.IsPresentInListSatisfying(supportedShell, processName, func(listElem string, toMatch string) bool {
+			return strings.HasPrefix(toMatch, listElem)
+		}) {
```

Comment:
note: if you switch the  2 arguments when invoking the callback in `IsPresentInListSatisfying`, I think this can be simplified to `crcstrings.IsPresentInListSatisfying(supportedShell, processName, strings.HasPrefix)`. Fine with me if you prefer to keep the code the way it is now.